cpp/dolfin/mesh/Topology.cpp

```diff
@@ -135,6 +135,54 @@ const std::vector<std::int32_t>& Topology::cell_owner() const
   return _cell_owner;
 }
 //-----------------------------------------------------------------------------
+std::vector<bool> Topology::surface_entity_marker(int dim) const
+{
+  const int tdim = this->dim();
+
+  if (dim >= tdim or dim < 0)
+  {
+    throw std::runtime_error("Invalid entity dimension: "
+                             + std::to_string(dim));
+  }
+
+  std::shared_ptr<const Connectivity> connectivity_facet_cell
```

Comment:
There should be a check that `connectivity_facet_cell` is not null. At least an assertion, but probably an exception should be thrown.

Reply:
:+1: